android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java

```diff
@@ -178,18 +182,35 @@ public void onDestroyView() {
 
     @Subscribe
     public void onLocationsDownloadDone(MicrolocationDownloadEvent event) {
-        if(swipeRefreshLayout == null)
-            return;
+        if (swipeRefreshLayout != null && swipeRefreshLayout.isRefreshing())
```

Comment:
this part seems repeating at more than one places. Have a utility method which takes in the refresh layout and does the action.